src/main/java/net/glowstone/command/minecraft/CloneCommand.java

```diff
@@ -0,0 +1,338 @@
+
+                        BlockEntity fromEntity = fromBlock.getBlockEntity();
+                        if (fromEntity != null) {
+                            BlockEntity toEntity = toBlock.getChunk().createEntity(toBlock.getX(), toBlock.getY(), toBlock.getZ(), toBlock.getTypeId());
+                            if (toEntity != null) {
+                                CompoundTag entityTag = new CompoundTag();
+                                fromEntity.saveNbt(entityTag);
+                                toEntity.loadNbt(entityTag);
+                            }
+                        }
+
+                        if (cloneMode == CloneMode.MOVE) {
+                            fromBlock.setType(Material.AIR, false);
+                        }
+
+                        blocksCloned++;
+                    }
+                }
+            }
+        }
+
+        sender.sendMessage("Cloned " + blocksCloned + " blocks.");
+        return true;
+    }
+
+    private static boolean between(double low, double high, double n) {
+        return low < n && n < high;
+    }
+
+    private static class ForwardsAxisIterable implements Iterable<Integer> {
+        private final int max;
+
+        private ForwardsAxisIterable(int max) {
+            this.max = max;
+        }
+
+        @NotNull
+        @Override
+        public Iterator<Integer> iterator() {
+            return new ForwardsAxisIterator(max);
+        }
+    }
+
+    private static class ForwardsAxisIterator extends AbstractIterator<Integer> {
+        private final int max;
+        private int current;
+
+        public ForwardsAxisIterator(int max) {
+            this.max = max;
+            this.current = 0;
+        }
+
+        @Override
+        protected Integer computeNext() {
+            if (current <= max) {
+                // Could use a post-increment operator here, but this is a bit more clear in getting the meaning across.
+                int retval = current;
+                current += 1;
+                return retval;
+            }
+            return endOfData();
+        }
+    }
+
+    private static class BackwardsAxisIterable implements Iterable<Integer> {
+        private final int max;
+
+        private BackwardsAxisIterable(int max) {
+            this.max = max;
+        }
+
+        @NotNull
+        @Override
+        public Iterator<Integer> iterator() {
+            return new BackwardsAxisIterator(max);
+        }
+    }
+
+    private static class BackwardsAxisIterator extends AbstractIterator<Integer> {
+        private int current;
+
+        public BackwardsAxisIterator(int length) {
+            this.current = length;
+        }
+
+        @Override
+        protected Integer computeNext() {
+            if (current >= 0) {
+                // Could use a post-decrement operator here, but this is a bit more clear in getting the meaning across.
+                int retval = current;
+                current -= 1;
```

Comment:
I understand why you'd prefer not using `int retval = current--;`, but maybe replace the `current -= 1` line with `current--;`

Reply:
Yeah, my biggest problem with the post- and pre- increment/decrement operators is that, in variable assignments, its not obvious which returns the old value and which returns the new value. I can definitely change that line to be current-- though.